flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroup.java

```diff
@@ -102,27 +104,53 @@
 
 	// ------------------------------------------------------------------------
 
+	@SuppressWarnings("unchecked")
```

Comment:
Should not be needed.

Reply:
is needed due to the unchecked assignment of `Map[]` to `Map<String, String>[]`

Reply:
Then I would suggest to limit the scope of the suppression to the variable assignment. One might need to factor out a method for this.